metadata/upload.go

```diff
@@ -260,6 +268,74 @@ type CachedToken struct {
 	ExpiresAt time.Time
 }
 
+// ######################################## //
+// block to implement buffered query into GCS
+type GcsQueryJob struct {
+	id    int
+	file *BQInput
+}
+type GcsQueryResult struct {
+	err   error
+	file *BQInput
+}
+func processGCSQuery(ctx context.Context, wg *sync.WaitGroup,
+					 jobs chan GcsQueryJob, results chan GcsQueryResult) {
+    for job := range jobs {
+		file := job.file
+		var e error = nil
```

Comment:
```suggestion
		var e error
```
`nil` is always the default value for reference types.